src/backend/columnar/columnar_customscan.c

```diff
@@ -825,8 +1400,40 @@ ColumnarScan_ExplainCustomScan(CustomScanState *node, List *ancestors,
 
 	if (scanDesc != NULL)
 	{
-		int64 chunkGroupsFiltered = ColumnarScanChunkGroupsFiltered(scanDesc);
-		ExplainPropertyInteger("Columnar Chunk Groups Removed by Filter", NULL,
-							   chunkGroupsFiltered, es);
+		CustomScan *cscan = castNode(CustomScan, node->ss.ps.plan);
+		List *chunkGroupFilter = cscan->custom_exprs;
+
+		if (chunkGroupFilter != NIL)
```

Comment:
To let users know, what about printing something like below if we had no chunk group filters:

```c
Columnar Chunk Group Filters: \<no useful quals found for chunk group filtering\>
Columnar Chunk Groups Removed by Filter: 0
```

Reply:
I think that's too verbose and not really following the pattern of other operations. We can add a debug GUC that prints extra information when planning, like `columnar.planner_debug_level = log` that defaults to `debug` or something.

Reply:
Added a debug GUC that prints extra information

Reply:
In explain pr (that I merged ~1 hour ago), we are doing something similar to that https://github.com/citusdata/citus/pull/5172/files#r699371509.
See https://github.com/citusdata/citus/blob/e41854f590213eb014f3bad2575fef499b78a703/src/test/regress/expected/columnar_chunk_filtering.out#L229.

Do you think we should also follow a pattern similar to what you've done here ?

Reply:
I think it makes sense for projection, because lack of projection means the optimization is being applied. Also, projection is simply more fundamental to the idea of a column store, so we should always have it there.

But lack of qual pushdown means it's not doing anything, so it makes sense to leave it out if there are no quals pushed down. Explain should be somewhat readable so I don't want to pollute it with every optimization that didn't happen.